ui/main/src/app/modules/cards/components/card/card.component.ts

```diff
@@ -55,7 +86,113 @@ export class CardComponent implements OnInit, OnDestroy {
+
+        console.log(`${this.lightCard.processInstanceId} =>  ${this.lightCard.lttd} :  ${this.handleDate(this.lightCard.lttd)}`);
+        if (this.entityUser !== undefined) {
+            if (!this.isArchivePageType() && this.lightCard.lttd != null
+                && this.lightCard.entitiesAllowedToRespond.includes(this.entityUser)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    isTimeToStartCountDown(): boolean {
+        const delta = this.calculateDiff();
+        return delta > 0 && delta < this.leftTime * 60;
+    }
+
+    startCountdownConfig() {
+
+        this.leftTime = this.configService.getConfigValue('feed.card.leftTime', false);
+        let leftTimeSeconds = this.leftTime * 60;
+        const endDate = new Date(this.calculateDiff()).getSeconds();
+
+        if (leftTimeSeconds > endDate) {
+            leftTimeSeconds = this.leftTime * 60;
+        } else {
+            leftTimeSeconds = endDate;
+        }
+
+        console.log(`startCountdownConfig >>>> ${this.lightCard.summary.key}  leftTime : ${leftTimeSeconds}`);
+
+        this.prettyConfig = {
+            leftTime: leftTimeSeconds,
+            format: 'mm:ss'
+        };
+
+        this.stopTime = false;
+        this.enableLastTimeToAct = true;
+        this.countdown.begin();
+    }
+
+
+    calculateDiff(): number {
+        const delta = Math.floor((this.lightCard.lttd - new Date().getTime()) / 1000);
+        console.log(`calculateDiff >>>>> ${this.lightCard.processInstanceId} : delta: ${delta} `);
+        return delta;
+    }
+
+    computeCountdown() {
+
+        if (!this.isValidatelttd()) {
+                console.log(`lttd not valid for :  ${this.lightCard.processInstanceId}`);
+                this.enableLastTimeToAct = false;
+                this.disabledActionButton();
+        } else {
+            console.log(`lttd valid for :  ${this.lightCard.processInstanceId}`);
+
+            if (this.calculateDiff() <= 0) {
+                console.log(`countDownOver ${this.lightCard.processInstanceId}`);
+                this.stopCountDown();
+            } else {
+
+                this.interval = setInterval(() => {
+                    if (this.isTimeToStartCountDown()) {
+                        this.startCountdownConfig();
+                        clearInterval(this.interval);
+                        return;
+                    }
+                }, 1000);
+            }
+        }
+    }
+
+    stopCountDown() {
+        this.stopTime = true;
+        this.enableLastTimeToAct = true;
+        this.disabledActionButton();
+        this.countdown.stop();
+    }
+
+    disabledActionButton() {
+        console.log(`disabledActionButton >>>>> for process card ${this.lightCard.processInstanceId}`);
+        this.store.dispatch(new UpdateLttd({ card: this.lightCard }));
+    }
+
+
+    handleEvent(e: CountdownEvent) {
+
+        if (this.calculateDiff() <= 0 && e.status === CountdownStatus.done) {
```

Comment:
Why make a calculateDiff  , is it not sufficient to have CountdownStatus.done ?

Reply:
No, that's enough, because the CountdownComponent component starts with a letTime at 0 and automatically sends an event down at the start.